src/diffusers/models/transformers/dit_transformer_2d.py

```diff
@@ -0,0 +1,268 @@
+        use_linear_projection (bool, optional, defaults to False):
+            Deprecated argument. Will be removed in a future version.
+        num_vector_embeds (bool, optional, defaults to False):
+            Deprecated argument. Will be removed in a future version.
+    """
+
+    _supports_gradient_checkpointing = True
+
+    @register_to_config
+    def __init__(
+        self,
+        num_attention_heads: int = 16,
+        attention_head_dim: int = 72,
+        in_channels: Optional[int] = None,
+        out_channels: Optional[int] = None,
+        num_layers: int = 28,
+        dropout: float = 0.0,
+        norm_num_groups: int = 32,
+        cross_attention_dim: Optional[int] = None,
+        attention_bias: bool = True,
+        sample_size: Optional[int] = None,
+        patch_size: Optional[int] = None,
+        activation_fn: str = "gelu-approximate",
+        num_embeds_ada_norm: Optional[int] = 1000,
+        only_cross_attention: bool = False,
+        upcast_attention: bool = False,
+        norm_type: str = "ada_norm_zero",
+        norm_elementwise_affine: bool = False,
+        norm_eps: float = 1e-5,
+        use_linear_projection: str = False,
+        num_vector_embeds: bool = False,
+    ):
+        super().__init__()
+
+        # Validate inputs.
+        deprecation_message = f"`use_linear_projection` is not a valid argument for {self.__class__.__name__}. Please update the configuration of this model to remove this argument. This argument will be removed from the class constructor in a future version."
+        deprecate("use_linear_projection", "1.0.0", deprecation_message)
+        deprecation_message = f"`num_vector_embeds` is not a valid argument for {self.__class__.__name__}. Please update the configuration of this model to remove this argument. This argument will be removed from the class constructor in a future version."
+        deprecate("num_vector_embeds", "1.0.0", deprecation_message)
+
+        if sample_size is None:
+            raise ValueError(f"{self.__class__.__name__} over patched input must provide `sample_size`.")
+
+        if in_channels is None and patch_size is None:
+            raise ValueError(
+                f"Has to define `in_channels`: {in_channels}, or patch_size:"
+                f" {patch_size}. Make sure that `in_channels`, and `patch_size` are not None."
+            )
+
+        if norm_type != "ada_norm_zero":
+            raise NotImplementedError(
+                f"Forward pass is not implemented when `patch_size` is not None and `norm_type` is '{norm_type}'."
+            )
+        elif norm_type == "ada_norm_zero" and num_embeds_ada_norm is None:
+            raise ValueError(
+                f"When using a `patch_size` and this `norm_type` ({norm_type}), `num_embeds_ada_norm` cannot be None."
+            )
+
+        # Set some common variables used across the board.
+        self.attention_head_dim = attention_head_dim
```

Comment:
it would be nice if we could remove these attributes e.g. `attention_head_dim`, `height`, `width`, `patch_size`
but I'm not sure if it would break anything like here https://github.com/huggingface/diffusers/issues/6838

Reply:
I think it might be better to keep them just to prevent #6838.